enf-reducer/promote-common-constraints.metta

```diff
@@ -0,0 +1,68 @@
+;(: promoteCommonConstraints (-> Expression Expression))
```

Comment:
Can you add a documentation here to help what promote common constraint does. Perhaps using some example expression too?